src/main/java/mrriegel/storagenetwork/block/cable/TileCable.java

```diff
@@ -1,665 +1,34 @@
-      compound.setString("south", south.toString());
-    if (east != null)
-      compound.setString("east", east.toString());
-    if (west != null)
-      compound.setString("west", west.toString());
-    if (up != null)
-      compound.setString("up", up.toString());
-    if (down != null)
-      compound.setString("down", down.toString());
-    NBTTagList nbttaglist = new NBTTagList();
-    for (int i = 0; i < upgrades.size(); ++i) {
-      if (upgrades.get(i) != null) {
-        NBTTagCompound nbttagcompound = new NBTTagCompound();
-        nbttagcompound.setByte("Slot", (byte) i);
-        upgrades.get(i).writeToNBT(nbttagcompound);
-        nbttaglist.appendTag(nbttagcompound);
-      }
-    }
-    compound.setTag("Items", nbttaglist);
     return compound;
   }
 
-  private boolean doesWrapperMatchStack(StackWrapper stackWrapper, ItemStack stack) {
-    ItemStack s = stackWrapper.getStack();
-    if (ores) {
-      return UtilTileEntity.equalOreDict(stack, s);
-    }
-    else if (metas) {
-      return s.getItem() == stack.getItem() && s.getItemDamage() == stack.getItemDamage();
-    }
-    else {
-      return stack.getItem() == s.getItem();
-    }
-    //    return ores ? UtilTileEntity.equalOreDict(stack, s) : metas ? stack.isItemEqual(s) : stack.getItem() == s.getItem();
-  }
-  /* key function used by TileMaster for all item trafic
-   * 
-   * TODO: TEST CASES
-   * 
-   * export + meta
-   * 
-   * export - meta
-   * 
-   * import + meta ; whitelist
-   * 
-   * import - meta ; whitelist
-   * 
-   * import + meta ; blacklist
-   * 
-   * import - meta ; blacklist */
-
-  //    TileMaster master;
-  private boolean doesPassOperationFilterLimit() {
-    TileMaster master = (TileMaster) this.world.getTileEntity(this.getMaster());
-    if (this.getUpgradesOfType(ItemUpgrade.OPERATION) < 1) {
-      return true;
-    }
-    //ok operation upgrade does NOT exist
-    //    TileMaster m = (TileMaster) world.getTileEntity(cable.getMaster());
-    if (getOperationStack() == null || getOperationStack().isEmpty()) {
-      return true;
-    }
-    int amount = master.getAmount(new FilterItem(getOperationStack()));
-    if (isOperationMode()) {
-      return amount > getOperationLimit();
-    }
-    else {
-      return amount <= getOperationLimit();
-    }
-  }
-
-  @Override
-  public boolean canTransfer(ItemStack stack, EnumFilterDirection way) {
-    if (!canGoThisDirection(way)) {
-      return false;
-    }
-    boolean filtered = checkWBFilterList(stack);
-    if (!filtered) {
-      return false;
-    }
-    //if thats ok, then also check 
-    if (!doesPassOperationFilterLimit()) {
-      return false;
-    }
-    return true;
-  }
-
-  private boolean canGoThisDirection(EnumFilterDirection way) {
-    if (isStorageEnabled() && !this.getTransferDirection().match(way)) {
-      return false;
-    }
-    return true;
-  }
-
-  private boolean checkWBFilterList(ItemStack stack) {
-    if (this.isWhitelist()) {
```

Comment:
This function is super ugly and gross.... but this is one of the places where the whitelist/blacklist feature of the cables comes into play

Reply:
Ok i see ItemStackMatcher  now, looks like that is replacing these features, it just needs to be hooked to GUI i assume?